homeassistant/components/google/config_flow.py

```diff
@@ -130,7 +135,22 @@ async def async_oauth_create_entry(self, data: dict) -> FlowResult:
             self.hass.config_entries.async_update_entry(entry, data=data)
             await self.hass.config_entries.async_reload(entry.entry_id)
             return self.async_abort(reason="reauth_successful")
-        return self.async_create_entry(title=self.flow_impl.name, data=data)
+
+        # Create a placeholder ConfigEntry to use since with the auth we've already created.
+        entry = ConfigEntry(version=1, domain=DOMAIN, title="", data=data, source="")
```

Comment:
Could we use the library authentication directly instead or another custom auth that doesn't require our OAuth2Session? Creating a config entry is hacky I think.

Reply:
Yes, very hacky. I'll create another custom auth.